wp-content/plugins/firebase-notifications/templates/settings.php

```diff
@@ -0,0 +1,36 @@
+<?php
```

Comment:
Why not remove the output buffering and print the html code in the `firebase_notification_settings_form()` function, instead of returning it to `firebase_notification_settings()`? As the `firebase_notification_settings()` does nothing else than printing out the return value, it might be better to just remove lines 1-3 and 34-36 and the keyword `echo` in [this line](https://github.com/Integreat/cms/pull/678/files#diff-6bbfdf482ccce06fda3c5da452532663R13). Same goes for the network setting page.

Reply:
Or even better: do not write the HTML into a variable at all. It can be directly printed in the file.

Reply:
This is actually what I tried to say :sweat_smile: